gnucash/report/reports/standard/category-barchart.scm

```diff
@@ -552,6 +569,7 @@ developing over time"))
                               (show-fullname? (gnc-account-get-full-name acct))
                               (else (xaccAccountGetName acct))))
                       (amounts (map gnc:gnc-monetary-amount (cadr series)))
+                      (percentage (map (lambda (x grandt)  (safe-/ x grandt)) amounts grandts))
```

Comment:
This line is ok for readability, but because `percentage` is only used iff `ratios-chart?` is `#t`, it would be marginally more efficient to generate the calls to `map` and `safe-/` only when required; thus:

```guile
  (gnc:html-chart-add-data-series!
   chart label (if ratio-chart? (map safe-/ amounts grandts) amounts) 
   color 'stack stack 'fill fill 'urls urls)
```